### ISSUE TITLE
# Bug: ODT templates not converting to PDFs in 7.0.3

### ISSUE THREAD

**Issue by SpWhatIsThisAbout** [Dolibarr/dolibarr#9224]
## Environment
- **Version**: 7.0.3
- **OS**: Linux XXXXXXX 4.15.0-29-generic #31-Ubuntu SMP Tue Jul 17 15:39:52 UTC 2018 x86_64
- **Web server**: Apache/2.4.29 (Ubuntu)
- **PHP**: 7.2.7-0ubuntu0.18.04.2
- **Database**: MySQL or MariaDB 5.7.22-0ubuntu18.04.1
- **URL(s)**: http://X.X.X.X/dolibarr/contrat/card.php?action=create&leftmenu=contracts
## Expected and actual behavior
Using an ODT template to create a contract or a customer order should result in a PDF.  I’ve loaded various ODT templates, those supplied with Dolibarr, another user as well as my own.  All result in the same behaviour/bug.  
The ODT template is renamed correctly and stored in the correct directory and contains the correct information. But with an .ODT extension and apparent structure which OpenOffice 5.1.5 regards as “corrupt” and “cannot be repaired”. 
(Example attached - (PROV14)_name_only.odt)
The logfile (attached dolibarr1.log) contains all the steps, uses the correct template, and creates the correct output directory etc.
Yet the logfile doesn’t seem to have an error message. 

## Steps to reproduce the behaviour
Starting with a clean installation of Dolibarr 7.0.3 
Upload 1nameonly.ODT (attached) into the correct directory 
Create a customer order using the odt template 
Generate a customer order 
Receive “The file was successfully generated” message 
Attempt to open the resulting file. 
Receive the “This file is corrupt” message 

Any clarifications required or suggested troubleshooting steps please let me know. All reasonable (and even zany) suggestions will be gladly received.   
## [Attached files]
[*Files*]
Attached: (1) ODT Template (2) ODT Output file (3) Logfile
[Archive.zip](https://github.com/Dolibarr/dolibarr/files/2270146/Archive.zip)

**Comment by SpWhatIsThisAbout** [comment]
Update to fault report:
I realise that most developers are focusing on the release of version 8, but I’ll record what I’ve done since anyhow.  
(a)	Totally wiped and reinstalled the server, fresh install of Ubuntu, PHP, MYSQL, Apache and Dolibarr. Using the latest version of each. 
(b)	Attempted to use a standard (proposal) ODT template – success!
(c)	Tried to use a customised ODT template – fail!
(d)	Back to the standard ODT template – fail!  And all subsequent attempts have also failed. 

To try to make some progress I created a new ODT template containing a single field {mycompany_name] and called it babysteps.otd (9kb in size) 
I used this as a template to create a contract and generate a corrupt file called (PROV4)_babysteps.odt (20kb in size) Also attached is the matching log file.  (dolibarr.1118) 

While (PROV4)_babysteps.odt is recognised as corrupt by OpenOffice, I can open it with a text editor and it contains the correct information.  
Using a free online tool (online.officerecovery.com) I can repair the corrupt file and it retains the correct information.  ((PROV4)_babysteps.OfficeRecovery Online Demo 3kb in size )

Use the oxygen xml editor (free trial version) I can see that the template has an internal structure of 3 toplevel directories: “Configuration2, META-INF & Thumbnails”. The corrupt output file has 4 toplevel directories: “Configuration2, META-INF, Pictures & Thumbnails”.  The repaired file has just 2 toplevel directories: “META-INF & Pictures”. There is a similar pattern with files.

It seems that somewhere spurious data is getting to the odt file which makes it unreadable. 

Attached in a zip file are:
(1)	Babysteps.odt – a very simple template 
(2)	(PROV4)_babysteps.odt – Dolibarr output file 
(3)	((PROV4)_babysteps.OfficeRecovery Online Demo.odt – repaired output file
(4)	Dolibarr.1118 - log file 
[Archive.zip](https://github.com/Dolibarr/dolibarr/files/2290071/Archive.zip)
